roles/apim-analytics-worker/tasks/main.yml

```diff
@@ -14,64 +14,61 @@
 #  limitations under the License.
 #----------------------------------------------------------------------------
 ---
-  - name: Set rpm_setup_path
-    set_fact:
-      setup_path: "{{ rpm_setup_path }}"
-      java_home: '"{{ rpm_setup_path }}/{{ product_name }}/{{ product_version }}/jdk/{{ java_version }}"'
-    when: ansible_os_family == "RedHat"
-
-  - name: Set deb_setup_path
-    set_fact:
-      setup_path: "{{ deb_setup_path }}"
-      java_home: '"{{ deb_setup_path }}/{{ product_name }}/{{ product_version }}/jdk/{{ java_version }}"'
-    when: ansible_os_family == "Debian"
-
   - block:
-    - name: Copy the Debian package of WSO2 API Manager Analytics
-      copy:
-        src: "{{ product_package_location }}/{{ product_archive_deb }}"
-        dest: "{{ product_path }}/{{product_archive_deb }}"
-        mode: 0755
-      when: ansible_os_family == "Debian"
-
-    - name: Copy the RPM package of WSO2 API Manager Analytics
-      copy:
-        src: "{{ product_package_location }}/{{ product_archive_rpm  }}"
-        dest: "{{ product_path  }}/{{ product_archive_rpm  }}"
-        mode: 0755
+    # JDK Distribution
+    - name: Set rpm_java_home
+      set_fact:
+        lib_dir: "{{ rpm_lib_dir }}"
+        java_home: "{{ rpm_lib_dir }}/{{ jdk_name }}"
       when: ansible_os_family == "RedHat"
 
-    - name: Install analytics .deb package
-      apt:
-        deb: "/opt/wso2/{{ product_archive_deb  }}"
+    - name: Set deb_java_home
+      set_fact:
+        lib_dir: "{{ deb_lib_dir }}"
+        java_home: "{{ deb_lib_dir }}/{{ jdk_name }}"
       when: ansible_os_family == "Debian"
 
-    - name: Install a .rpm package
-      yum:
-        name: "{{ product_path  }}/{{ product_archive_rpm  }}"
-      when: ansible_os_family == "RedHat"
+    - name: Create lib directory
+      file:
+        path: "{{ lib_dir }}"
+        state: directory
 
-    - name: "Copy JDBC Driver"
-      copy:
-       src: "files/{{ jdbc_driver }}"
-       dest: "{{ setup_path }}/{{ product_name }}/{{ product_version }}/lib"
+    - name: Unarchive JDK distribution
+      unarchive:
+        src: "{{ product_package_location }}/{{ jdk_name }}.tar.gz"
+        dest: "{{ lib_dir }}"
 
-    - name: "Copy deployment.yaml"
-      template:
-        src: "carbon-home/conf/worker/deployment.yaml.j2"
-        dest: "{{ setup_path }}/{{ product_name }}/{{ product_version }}/conf/worker/deployment.yaml"
+    # WSO2 Distribution
+
+    - name: Create distribution path
+      file:
+        path: "{{ distribution_path }}"
+        state: directory
+        mode: 0755
+
+    - name: Unzip WSO2 Identity Server Package
```

Comment:
Should be API Manager

Reply:
Fixed https://github.com/wso2/ansible-apim/pull/21/commits/cbf501d33b78acaf4e80694eb13ce8b2b43991e9